### PR TITLE
Support Azure OpenAI models DATANG-3500

### PR DESCRIPTION
**Pull request by samsucik**
Jira: DATANG-3500

As part of this change, I decided to also update the version of `openai` we use. It was much simpler than dealing with the old version where one had to directly set attributes of the imported `openai` module in order to use the Azure OpenAI models.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,14 +43,17 @@ Create a virtual environment that uses Python 3.10. Then, install project requir
...
+- `AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_API_BASE`: Optional. Only if you want to use OpenAI
+  models (ChatGPT, GPT-4, etc.) via Azure OpenAI APIs.
 - `PROMPTERATOR_DATA_DIR`: Optional. Where to store the files with your prompts and generated
   texts. Defaults to `~/prompterator-data`. If you plan to work on prompts for different tasks
   or datasets, it's a good idea to use a separate directory for each one.
 
-If you do not happen to have access to `OPENAI_API_KEY`, feel free to use the
-`mock-gpt-3.5-turbo` model, which is a mocked version of the OpenAI's GPT-3.5
-model. This is also very helpful when developing Prompterator itself.
+If you do not happen to have access to an OpenAI API key, feel free to use the `mock-gpt-3. 5-turbo`
```
Comment (DanielSkala) [inline README.md]:
```suggestion
If you do not happen to have access to an OpenAI API key, feel free to use the `mock-gpt-3.5-turbo`
```

File: poetry.lock
Comment (DanielSkala) [inline poetry.lock:1]: Hmm, it seems that many other dependencies were changed as well but it works for me well. Have you ran the lock with `--no-update`?

**Review by DanielSkala — APPROVED** [review]
Looks good to me @samsucik - also works nicely! 

I only had one question about the poetry lock but if you ran also `--no-update` I think we can merge it :)

File: poetry.lock
Comment (samsucik) [reply poetry.lock:1]: @DanielSkala as strange as it seems, I think it's OK. When I now run `poetry lock --no-update`, it generates the very same lock file (based on the updated version of `pyproject.toml` with the newer `openai` version).

**Pull request merged**